### PR TITLE
[feature](show-frontends-disk) Add Show frontend disks

### PR DESCRIPTION
**Pull request by herry2038**
## Proposed changes

Issue Number: close #xxx

Add feature show frontends disk which can show the disk information of fe 's interested dir, such as doris-meta, log, audit-log and temp.
As a example:

`mysql> show frontends disks;
+-----------------------------------------+------------+-------------+-----------+-----------------------------+------------+----------+------+-----------+---------+---------+
| Name                                    | Host       | EditLogPort | DirType   | Dir                         | Filesystem | Capacity | Used | Available | UseRate | MountOn |
+-----------------------------------------+------------+-------------+-----------+-----------------------------+------------+----------+------+-----------+---------+---------+
| fe_6a06ffaa_8346_4381_b462_8e33209ca6ec | 172.1.1.1  | 9010        | meta      | E:\data\doris\fe/doris-meta |            | 0        | 0    | 8191P     | 0%      |         |
| fe_6a06ffaa_8346_4381_b462_8e33209ca6ec | 172.1.1.1  | 9010        | log       | E:\data\doris\fe/log        |            | 0        | 0    | 8191P     | 0%      |         |
| fe_6a06ffaa_8346_4381_b462_8e33209ca6ec | 172.1.1.1  | 9010        | audit-log | E:\data\doris\fe/log        |            | 0        | 0    | 8191P     | 0%      |         |
| fe_6a06ffaa_8346_4381_b462_8e33209ca6ec | 172.1.1.1  | 9010        | temp      | E:\data\doris\fe/temp_dir   |            | 0        | 0    | 8191P     | 0%      |         |
| fe_fe1d5bd9_d1e5_4ccc_9b03_ca79b95c9941 | 172.1.1.2  | 9010        | meta      | /data/doris/fe/doris-meta   | /dev/sdc5  | 366G     | 111G | 236G      | 33%     | /data   |
| fe_fe1d5bd9_d1e5_4ccc_9b03_ca79b95c9941 | 172.1.1.2  | 9010        | log       | /data/doris/fe/log          | /dev/sdc5  | 366G     | 111G | 236G      | 33%     | /data   |
| fe_fe1d5bd9_d1e5_4ccc_9b03_ca79b95c9941 | 172.1.1.2  | 9010        | audit-log | /data/doris/fe/log          | /dev/sdc5  | 366G     | 111G | 236G      | 33%     | /data   |
| fe_fe1d5bd9_d1e5_4ccc_9b03_ca79b95c9941 | 172.1.1.2  | 9010        | temp      | /data/doris/fe/temp_dir     | /dev/sdc5  | 366G     | 111G | 236G      | 33%     | /data   |
+-----------------------------------------+------------+-------------+-----------+-----------------------------+------------+----------+------+-----------+---------+---------+
8 rows in set (0.00 sec)`

### CODE REVIEW

**Comment by herry2038** [comment]
run buildall

**Review by chenlinzhong — DISMISSED** [review]
LGTM

**Comment by herry2038** [comment]
run buildall

**Comment by hello-stephen** [comment]
(From new machine)TeamCity pipeline, clickbench performance test result:
 the sum of best hot time: 44.58 seconds
 stream load tsv:          503 seconds loaded 74807831229 Bytes, about 141 MB/s
 stream load json:         21 seconds loaded 2358488459 Bytes, about 107 MB/s
 stream load orc:          65 seconds loaded 1101869774 Bytes, about 16 MB/s
 stream load parquet:          31 seconds loaded 861443392 Bytes, about 26 MB/s
 insert into select:          28.8 seconds inserted 10000000 Rows, about 347K ops/s
 storage size: 17161672703 Bytes

**Comment by herry2038** [comment]
run buildall

**Comment by hello-stephen** [comment]
(From new machine)TeamCity pipeline, clickbench performance test result:
 the sum of best hot time: 46.39 seconds
 stream load tsv:          506 seconds loaded 74807831229 Bytes, about 140 MB/s
 stream load json:         20 seconds loaded 2358488459 Bytes, about 112 MB/s
 stream load orc:          65 seconds loaded 1101869774 Bytes, about 16 MB/s
 stream load parquet:          31 seconds loaded 861443392 Bytes, about 26 MB/s
 insert into select:          29.1 seconds inserted 10000000 Rows, about 343K ops/s
 storage size: 17162395358 Bytes

**Comment by herry2038** [comment]
run p0

**Review by chenlinzhong — DISMISSED** [review]
LGTM

File: fe/fe-common/src/main/java/org/apache/doris/common/io/DiskUtils.java [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+                1L * 1024,
+                1L * 1024 * 1024,
+                1L * 1024 * 1024 * 1024,
+                1L * 1024 * 1024 * 1024 * 1024,
+                1L * 1024 * 1024 * 1024 * 1024 * 1024,
+                1L * 1024 * 1024 * 1024 * 1024 * 1024 * 1024,
+        };
+
+        for (long v : values) {
+            System.out.println(sizeFormat(v));
```
Comment (luozenglin) [inline fe/fe-common/src/main/java/org/apache/doris/common/io/DiskUtils.java:116]: Is this for debugging? Please add unit tests or regression tests

**Comment by luozenglin** [comment]
Please add documentation. You can also add the tvf function if you are interested, ref: #20857

File: fe/fe-common/src/main/java/org/apache/doris/common/io/DiskUtils.java (hunk 1 above)
Comment (herry2038) [reply fe/fe-common/src/main/java/org/apache/doris/common/io/DiskUtils.java]: Add a unit test class.

**Comment by herry2038** [comment]
run buildall

**Comment by hello-stephen** [comment]
(From new machine)TeamCity pipeline, clickbench performance test result:
 the sum of best hot time: 45 seconds
 stream load tsv:          506 seconds loaded 74807831229 Bytes, about 140 MB/s
 stream load json:         20 seconds loaded 2358488459 Bytes, about 112 MB/s
 stream load orc:          65 seconds loaded 1101869774 Bytes, about 16 MB/s
 stream load parquet:          31 seconds loaded 861443392 Bytes, about 26 MB/s
 insert into select:          29.2 seconds inserted 10000000 Rows, about 342K ops/s
 storage size: 17161436661 Bytes

**Comment by herry2038** [comment]
> Please add documentation. You can also add the tvf function if you are interested, ref: #20857



> Please add documentation. You can also add the tvf function if you are interested, ref: #20857

已经提交了文档了。tvf函数晚点在实现，也可以等下一个版本。

**Comment by herry2038** [comment]
run buildall

File: fe/fe-core/src/main/java/org/apache/doris/service/ExecuteEnv.java [hunk 2]
```diff
@@ -28,10 +33,55 @@ public class ExecuteEnv {
     private ConnectScheduler scheduler;
     private long startupTime;
 
+    public static class DiskInfo {
+        private String dirType;
```
Comment (luozenglin) [inline fe/fe-core/src/main/java/org/apache/doris/service/ExecuteEnv.java:37]: This class is used elsewhere and should be put in a class file, with `toThrift` and `fromThrift` as member method

File: fe/fe-core/src/main/java/org/apache/doris/service/ExecuteEnv.java [hunk 3]
```diff
@@ -56,4 +106,55 @@ public MultiLoadMgr getMultiLoadMgr() {
     public long getStartupTime() {
         return startupTime;
     }
+
+    public List<DiskInfo> getDiskInfos() {
+        return this.diskInfos;
+    }
+
+    public List<DiskInfo> refreshAndGetDiskInfo(boolean refresh) {
+        for (DiskInfo disk : diskInfos) {
```
Comment (luozenglin) [inline fe/fe-core/src/main/java/org/apache/doris/service/ExecuteEnv.java:115]: Where do followers call this method to update the disk info?

**Comment by herry2038** [comment]
run buildall

**Comment by herry2038** [comment]
run buildall

**Comment by herry2038** [comment]
run buildall

**Comment by hello-stephen** [comment]
(From new machine)TeamCity pipeline, clickbench performance test result:
 the sum of best hot time: 47.73 seconds
 stream load tsv:          515 seconds loaded 74807831229 Bytes, about 138 MB/s
 stream load json:         20 seconds loaded 2358488459 Bytes, about 112 MB/s
 stream load orc:          65 seconds loaded 1101869774 Bytes, about 16 MB/s
 stream load parquet:          30 seconds loaded 861443392 Bytes, about 27 MB/s
 insert into select:          29.8 seconds inserted 10000000 Rows, about 335K ops/s
 storage size: 17162380869 Bytes

**Review by luozenglin — APPROVED** [review]
LGTM

**Review by chenlinzhong — APPROVED** [review]
LGTE

**Pull request merged**